Mathlib/Algebra/CharP/Basic.lean

```diff
@@ -605,7 +605,12 @@ theorem ringChar_of_prime_eq_zero [Nontrivial R] {p : ℕ} (hprime : Nat.Prime p
   Or.resolve_left ((Nat.dvd_prime hprime).1 (ringChar.dvd hp0)) ringChar_ne_one
 #align char_p.ring_char_of_prime_eq_zero CharP.ringChar_of_prime_eq_zero
 
-end CharOne
+theorem charP_iff_prime_eq_zero [Nontrivial R] {p : ℕ} (hp : p.Prime) :
```

Comment:
I'd be tempted to add a `cast` to this name to match `cast_eq_zero`, though you should do the same to the `ringChar` lemma above (and the PR title) if you agree:
```suggestion
theorem charP_iff_cast_eq_zero_of_prime [Nontrivial R] {p : ℕ} (hp : p.Prime) :
```
bors d+